send-slack-notification/action.yaml

```
@@ -15,17 +15,42 @@ runs:
   using: composite
   steps:
     - name: Retrieve Slack Thread ID
-      if: ${{ github.run_attempt > 1 }}
+      id: retrieve-slack-thread-id
+      continue-on-error: true
       uses: actions/download-artifact@d3f86a106a0bac45b974a628896c90dbdf5c8093
       with:
         name: slack-thread-id-${{ github.run_id }}
 
-    - name: Output Slack Thread ID
-      id: slack-thread-id
-      if: ${{ github.run_attempt > 1 }}
+    - name: Extract Slack Thread ID into Environment Variable
+      env:
+        RETRIEVE_SLACK_THREAD_ID_OUTCOME: ${{ steps.retrieve-slack-thread-id.outcome }}
+      shell: bash
+      run: |
+        if [ "$RETRIEVE_SLACK_THREAD_ID_OUTCOME" = "success" ]; then
+          echo "SLACK_THREAD_ID=$(cat slack-thread-id)" | tee -a "$GITHUB_ENV"
+        else
+          echo "SLACK_THREAD_ID=none" | tee -a "$GITHUB_ENV"
+        fi
```

This can be simplified (requiring a change further down):

```suggestion
    - name: Extract Slack Thread ID into Environment Variable
      if: steps.retrieve-slack-thread-id.outcome == 'success'
      shell: bash
      run: |
        echo "SLACK_THREAD_ID=$(cat slack-thread-id)" | tee -a "$GITHUB_ENV"
```